lib/src/Uncertainty/Distribution/Mixture.cxx

```diff
@@ -603,6 +603,25 @@ NumericalSample Mixture::getSupport(const Interval & interval) const
   return support;
 }
 
+/* Get the PDF singularities inside of the range - 1D only */
+NumericalPoint Mixture::getSingularities() const
+{
+  if (getDimension() > 1) throw InternalException(HERE) << "Error: getSingularities() is defined for 1D distributions only";
+  NumericalPoint singularities(0);
+  // Aggregate all the singularities of the atoms including the bounds of the range
+  for (UnsignedInteger i = 0; i < distributionCollection_.getSize(); ++i)
+    {
+      singularities.add(distributionCollection_[i].getRange().getLowerBound()[0]);
+      singularities.add(distributionCollection_[i].getSingularities());
+      singularities.add(distributionCollection_[i].getRange().getUpperBound()[0]);
+    }
+  // Sort the singularities
+  std::sort(singularities.begin(), singularities.end());
+  NumericalPoint result(singularities.getDimension() - 2);
```

Comment:
Why do you put aside the first and last ones?
We way also face to duplicates....